### PR TITLE
Added cloud-config headers explanation

### PR DESCRIPTION
**Pull request by kayrus**
/cc @robszumski

### CODE REVIEW

File: Documentation/cloud-config.md [hunk 1]
```diff
@@ -26,7 +26,7 @@ We've designed our implementation to allow the same cloud-config file to work ac
 
 The cloud-config file uses the [YAML][yaml] file format, which uses whitespace and new-lines to delimit lists, associative arrays, and values.
 
-A cloud-config file must contain `#cloud-config`, followed by an associative array which has zero or more of the following keys:
+A cloud-config file must contain `#cloud-config` header or header starts on `#!`. If cloud-config has `#cloud-config` header, it should followed by an associative array which has zero or more of the following keys:
```
Comment (robszumski) [inline Documentation/cloud-config.md]: This reads a bit confusing to me. What about:

"A cloud-config file must contain a header: either `#cloud-config` for processing as cloud-config (suggested) or `#!` for processing as a shell script (advanced). If cloud-config has `#cloud-config` header, it should followed by an associative array which has zero or more of the following keys:"

File: Documentation/cloud-config.md [hunk 2]
```diff
@@ -37,6 +37,8 @@ A cloud-config file must contain `#cloud-config`, followed by an associative arr
 
 The expected values for these keys are defined in the rest of this document.
 
+If cloud-config header starts on `#!` then coreos-cloudinit will recognize it as bash script and run it as transient systemd service.
```
Comment (robszumski) [inline Documentation/cloud-config.md]: Is this technically a bash script or a shell script?

Comment (crawford) [inline Documentation/cloud-config.md]: It is a shell script which is interpreted by bash on CoreOS.

**Comment by kayrus** [comment]
Updated

**Comment by robszumski** [comment]
LGTM

**Pull request merged**